### PR TITLE
Fix/add prettier lint

### PR DESCRIPTION
**Pull request by jsdalton**
## Purpose

Sometimes code sneaks in that avoids a prettier (re)write. (Not sure why, it might have to do with some folks using a Git client that skips the precommit hooks 🤷 ).

Ideally we'd like to fail the build if the branch contains changes that should have been rewritten by prettier. This ensures that code introduced to the repo is always in the correct "pretty" shape.

## Approach

* Write a shell script that runs against the change set represented by the branch
* Test against `origin/master`. Unfortunately there's no easy way to identify the "target branch" in our circle environment, so this _could_ mean the check runs against a larger surface area then intended, e.g. if you have a branch of a branch.
* Added a kill switch `$SKIP_PRETTIER_CHECK` env var we can set in the build settings to turn this off if it acts up (since it might be flakier than we expect!)

> Note: See builds in the commit history for some examples of failure modes when problems are found.

### CODE REVIEW

File: .circleci/prettier-check.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#! /bin/sh
+# Runs a prettier check against changed files on the current branch compared against master.The
+# purpose of this script is to fail if a change gets passed the commit hooks which enforce prettier
+# rules.
+
+# kill switch if for some reason this script inadvertently blocks the build
+test -n "$SKIP_PRETTIER_CHECK" && exit 0
```
Comment (ghepting) [inline .circleci/prettier-check.sh:7]: 👌 good call :)

**Review by ghepting — APPROVED** [review]

**Review by kdamball — COMMENTED** [review]
Is there a difference between this and what is happening here: https://github.com/contentful/apps/blob/26c4360d49a0810a9ce6f76ab094989807409893/.circleci/config.yml#L42-L44 ?

File: .prettierrc [hunk 2]
```diff
@@ -3,6 +3,6 @@
   "tabWidth": 2,
   "useTabs": false,
   "singleQuote": true,
-  "jsxBracketSameLine": true
+  "bracketSameLine": true
```
[the remark this answers is not in the document]
Comment (jsdalton) [reply .prettierrc:6]: This was actually caught by the new check we introduced. I consider it a good thing! 🙂

**Review by mgoudy91 — APPROVED** [review]

**Comment by jsdalton** [comment]
@kdamball Yes, the lint just runs `eslint`, which checks a bunch of eslint rules but doesn't do verify any of the prettier rules.

So for example, prettier enforces `;` at the end of lines. eslint doesn't care. As a result, you could (if you bypass precommit hooks) introduce a commit without the `;` at the end of a line and the lint would happily pass. We had this happen recently, and then @ghepting introduced a change with bunch of prettier fixes that were totally unrelated to his work!

The idea here is just to do a very lightweight check to make sure the prettier hook actually ran and to fail swiftly if it did not, and avoid that in the future. Also since it's a project wide task vs. the lint (which uses lerna and is run package by package) it made sense to keep separate.

**Pull request merged**